File: languages/go/entity/identityV2/date_attribute.go
package identityV2

import (
	"time"

	"github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"
)

type DateAttribute struct {
	Attribute
}

func NewDateAttribute(key string, value time.Time) DateAttribute {
	a := Attribute{
		Id:    key,
		Value: value.Format("2006-01-02"),
	}
	return DateAttribute{
		Attribute: a,
	}
}

func NewDateAttributeFromProto(s *proto.DateAttributeV2) DateAttribute {
	if s == nil {
		return DateAttribute{}
	}
	a := Attribute{
		Id:    s.GetId(),
		Value: s.GetValue(),
	}
	return DateAttribute{
		Attribute: a,
	}
}

func (s DateAttribute) ToProto() *proto.DateAttributeV2 {
	return &proto.DateAttributeV2{
		Id:    s.Attribute.Id,
		Value: s.Attribute.Value.(string),
	}
}

File: languages/python/bloock/entity/identity_v2/date_attribute.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute import Attribute


class DateAttribute(Attribute):
    @staticmethod
    def from_proto(a: proto.DateAttributeV2) -> DateAttribute:
        return DateAttribute(
            id=a.id,
            value=a.value,
        )

    def to_proto(self) -> proto.DateAttributeV2:
        return proto.DateAttributeV2(
            id=self.id,
            value=self.value,
        )